Source/WebKit/UIProcess/API/glib/WebKitWebsiteDataManager.cpp

```diff
@@ -479,6 +490,41 @@ static void webkit_website_data_manager_class_init(WebKitWebsiteDataManagerClass
             nullptr, nullptr,
             FALSE,
             static_cast<GParamFlags>(WEBKIT_PARAM_READWRITE | G_PARAM_CONSTRUCT_ONLY)));
+
+    /**
+     * WebKitWebsiteDataManager:origin-storage-ratio:
+     *
+     * The ratio which is used to calculate the origin data storage based on disk space.
+     * The origin data storage if it is exceeded, request will fail.
+     * Value of 0.0 means that data storage is not allowed.
+     *
+     * Since: 2.42
+     */
+    g_object_class_install_property(
+        gObjectClass,
+        PROP_ORIGIN_STORAGE_RATIO,
+        g_param_spec_double("origin-storage-ratio",
+            nullptr, nullptr,
+            0.0, 1.0, 0.6,
+            static_cast<GParamFlags>(WEBKIT_PARAM_WRITABLE | G_PARAM_CONSTRUCT_ONLY)));
```

Comment:
Since this is a construct property it means it will always be set to the default value on construction. I think we should keep current behavior when these properties are not set. Since we don't have optional, I think we can use -1 as the default behavior. So, I would use -1 as the min and default values, and 1 as the maximum.